### PR TITLE
added std::begin/end

### PR DESCRIPTION
**Pull request by muhammetasan**
I added explanation about begin/end free functions introduced in C++11

### CODE REVIEW

**Comment by mguid65** [comment]
You seem to have intended to put a / in the section title but it is not there and says:

# std::beginend

I would remove the explanation about containers begin and end functions along with the first code snippet.

`std::begin` and `std::end` free functions were added to return begin and end iterators of a container generically. These functions also work with raw arrays which do not have `begin` and `end` member functions.

```c++
template <typename T>
int CountTwos(const T& container) {
  return count_if(std::begin(container), std::end(container), [](int item) {
    return item == 2;
  });
}

vector<int> vec = {2,2,43,435,4543,534};
int arr[8] = {2,43,45,435,32,32,32,32};
auto a = CountTwos(vec); // 2
auto b = CountTwos(arr);  // 1
```

**Comment by muhammetasan** [comment]
thanks for suggestions.

File: CPP11.md [hunk 1]
```diff
@@ -874,6 +875,25 @@ auto handle = std::async(std::launch::async, foo);  // create an async task
 auto result = handle.get();  // wait for the result
 ```
 
+### std::begin/end
+`std::begin` and `std::end` free functions were added to return begin and end iterators of a container generically. These functions also work with raw arrays which do not have begin and end member functions.
+
+```c++
+template <typename T>
+int CountTwos(const T& container) {
+  return count_if(std::begin(container), std::end(container), [](int item) {
```
Comment (AnthonyCalandra) [inline CPP11.md]: Use `std::count_if`.

Also, even though we are just counting on `int` elements, I think we should still make the lambda parameter a `auto&` to keep the code as generic as we can. e.g. Think of if we were given a container with custom objects.

File: CPP11.md [hunk 2]
```diff
@@ -874,6 +875,25 @@ auto handle = std::async(std::launch::async, foo);  // create an async task
...
+    return item == 2;
+  });
+}
+
+vector<int> vec = {2,2,43,435,4543,534};
+int arr[8] = {2,43,45,435,32,32,32,32};
+auto a = CountTwos(vec); // 2
+auto b = CountTwos(arr);  // 1
+
```
Comment (AnthonyCalandra) [inline CPP11.md]: Extra space.

File: CPP11.md [hunk 3]
```diff
@@ -874,6 +875,25 @@ auto handle = std::async(std::launch::async, foo);  // create an async task
...
+```
+
```
Comment (AnthonyCalandra) [inline CPP11.md]: Extra space.

File: CPP11.md [hunk 4]
```diff
@@ -874,6 +875,25 @@ auto handle = std::async(std::launch::async, foo);  // create an async task
...
+```c++
+template <typename T>
+int CountTwos(const T& container) {
+  return count_if(std::begin(container), std::end(container), [](int item) {
+    return item == 2;
+  });
+}
+
+vector<int> vec = {2,2,43,435,4543,534};
```
Comment (AnthonyCalandra) [inline CPP11.md]: `std::vector<int>`

File: CPP11.md (hunk 1 above)
Comment (muhammetasan) [reply CPP11.md]: This is a c++11 example code. Generic lambda parameters are introduced in C++14.

Comment (AnthonyCalandra) [reply CPP11.md]: You are correct! Good catch.

**Pull request merged**